substrate/client/network/src/request_responses.rs

```diff
@@ -80,6 +114,8 @@ pub enum RequestFailure {
 	Obsolete,
 	#[error("Problem on the network: {0}")]
 	Network(OutboundFailure),
+	#[error("Problem on the network: {0}")]
+	Network2(CustomOutboundFailure),
```

Comment:
We should replace the `Network` variant

Reply:
Yes, we need to have only one `Network` variant (we can then use `Network` name for your `Network2`). This will make runtime checks like `RequestFailure::Network(_) => {debug_assert!(false, "Don't use deprecated Network");}` unneeded and the code safer.

You are free to change other files as well. Let us know if there is too much to change.

Reply:
> Yes, we need to have only one `Network` variant (we can then use `Network` name for your `Network2`). This will make runtime checks like `RequestFailure::Network(_) => {debug_assert!(false, "Don't use deprecated Network");}` unneeded and the code safer.
> 
> You are free to change other files as well. Let us know if there is too much to change.

I gave it another try, replaced/removed the variant `Network` (I still call the new variant Network2 to make it obvious...), but there was a price to pay because of the use of `request_response::Event::OutboundFailure` and `request_response::Event::InboundFailure` in request_responses.rs. The (Temporary I hope) solution I found is to point at specific error to return, instead of having a more adaptive behaviour.

Replacing these two libp2p dependent Events is the next step, in order to restore the original error behavior.